### PR TITLE
Publisher Specification

### PR DESCRIPTION
**Pull request by ryanml**
Progress for the Publishers' specification

### CODE REVIEW

File: js/publisher.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      PAYMENTS_ENABLED: true,
+      PAYMENTS_NOTIFICATION_TRY_PAYMENTS_DISMISSED: true
+    }
+  }
+
+  setState (newState) {
+    let oldState = this.state
+    this.state = newState
+
+    if (!newState.get('ledger').hasIn(['info'])) {
```
Comment (NejcZdovc) [inline js/publisher.js]: can be simplified into `newState.hasIn(['ledger', 'info'])`

File: js/publisher.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  constructor () {
+    super()
+    this.ledger = null
+    this.timeStamp = 1525688397657
+    this.state = defaultAppState
+    this.mediaType = null
+    this.mediaMinimum = null
+  }
+
+  get settingsP () {
```
Comment (NejcZdovc) [inline js/publisher.js]: I see that this function is used in other tests as well. Can we pull it in shared?

File: abs/index.js [hunk 3]
```diff
@@ -20,6 +80,43 @@ class Lib {
...
+    if (!oldState.hasIn(['ledger', 'info'])) {
+      oldState = oldState.setIn(['ledger', 'info'], Immutable.Map())
+    }
+
+    const newInfo = newState.getIn(['ledger', 'info'])
+
+    newInfo.keySeq().forEach((key) => {
+      const temp = newInfo.getIn([key])
+
+      if (Immutable.Map.isMap(temp) &&
```
Comment (NejcZdovc) [inline abs/index.js]: `if (Immutable.Map.isMap(temp) && temp.size === 0) {` can we do it like this?

Comment (ryanml) [reply abs/index.js]: Definitely :) This was an oversight for sure

**Pull request merged**